requirements/dev.txt

```diff
@@ -0,0 +1,7 @@
+black==22.3.0
+isort==5.10.1
+pylint
+flake8
+pytest
+pre-commit>=2.15.0
+tox>=3.24.3
```

Comment:
It might be an idea to fix the versions to the ones used in pre-commit [configurations](https://github.com/ORippler/anomalib/blob/fix_docker/.pre-commit-config.yaml). 

```yaml
black==22.6.0
isort==5.10.1
pylint==2.14.5
flake8==4.0.1
pytest
```

There are also other tools used in pre-commit such as mypy, pydocstyle, nbQA. Should we include them as well?

Reply:
I would only include those tools that are [used by vscode](https://github.com/openvinotoolkit/anomalib/blob/main/.devcontainer/devcontainer.json#L10-L27) during programming/on every save, and pin them to the revs used in pre-commit.

While one could also run mypy etc. on save, this is a bit too strict/restrictive imo.